src/components/Usermenu.tsx

```diff
@@ -0,0 +1,201 @@
+ */
+type UsermenuProps = {
+  anchorElement: null | HTMLElement;
+  userPopUpOpen: boolean;
+  setUserPopUpOpen: (open: boolean) => void;
+  setAnchorElement: (anchorEl: null | HTMLElement) => void;
+};
+
+/**
+ * UserPopup Component
+ *
+ * This component renders a User Popup which appears when the user clicks on the
+ * account icon in the header. It provides options to view the user's profile,
+ * navigate to the dashboard, and log out. It also displays the current app version.
+ *
+ * @param {UsermenuProps} props - The properties of the Usermenu component
+ * @returns {ReactElement} A Usermenu component
+ *
+ */
+const Usermenu = ({
+  anchorElement,
+  userPopUpOpen,
+  setUserPopUpOpen,
+  setAnchorElement,
+}: UsermenuProps): ReactElement => {
+  const theme = useTheme();
+  const { isDownXs, isBetweenXsSm} = useBreakpoints();
+  const placeholderUser = usePlaceholder();
+
+  /**
+   * Function to handle closing the user popup
+   */
+  const handleClose = (): void => {
+    setUserPopUpOpen(false);
+    setAnchorElement(null);
+  };
+
+  /**
+   * Prevents the default behavior of the event
+   */
+  const preventDefault = (event: React.MouseEvent<HTMLLIElement, MouseEvent>): void => {
+    event.preventDefault();
+  };
+
+  window.addEventListener('resize', handleClose);
+
+  return (
+    <Menu
+      anchorEl={anchorElement}
+      id="account-menu"
+      open={userPopUpOpen || false}
+      onClose={handleClose}
+      onClick={handleClose}
+      data-testid="user-menu"
+      transitionDuration={0}
+      keepMounted
+      PaperProps={{
+        elevation: 0,
+        sx: {
+          overflow: "visible",
+          filter: "drop-shadow(0px 2px 8px rgba(0,0,0,0.32))",
+          mt: 1.5,
+          bgcolor: theme.palette.secondary.main,
+          color: theme.palette.text.secondary,
+          minWidth: 120,
+          border: "1.5px solid",
+          borderColor: theme.palette.text.primary,
+          "&::before": {
+            content: '""',
+            display: "block",
+            position: "absolute",
+            borderLeft: "1.5px solid",
+            borderTop: "1.5px solid",
+            borderColor: theme.palette.text.primary,
+            top: 0,
+            transform: "translateY(-54%) rotate(45deg)",
+            zIndex: 0,
+            bgcolor: theme.palette.secondary.main,
+            width: 15,
+            height: 15,
+            left: 'calc(88% - 7.5px)', // Adjusting lozenge to center it
+          },
+        },
+      }}
+      transformOrigin={{ horizontal: "right", vertical: "top" }}
+      anchorOrigin={{ horizontal: "right", vertical: "bottom" }}
+    >
+      <MenuItem
+      sx={{
+        minWidth: 36,
+        borderBottom: '1px solid #043f5f', // Placeholder for divider since divider have an unremovable margin
+        ':hover': {
+          backgroundColor: 'transparent',
+        },
+        cursor: 'default',
+      }}
+      onClick={preventDefault}
+    >
```

Comment:
instead of defining the function above since you are only using this in one place `onClick={ (event) => event.preventdefault() }`